### PR TITLE
Some refactorings

### PR DESCRIPTION
**Pull request by alonsogarciapablo**
@xavijam PTAL

### CODE REVIEW

File: src/geo/layer_definition.js [hunk 1]
```diff
@@ -2,21 +2,10 @@
 function MapBase(options) {
```
Comment (alonsogarciapablo) [inline src/geo/layer_definition.js]: This file should not be used anymore, so we should be able to delete it pretty soon.

File: src/windshaft/client.js [hunk 2]
```diff
@@ -26,25 +26,22 @@ cdb.windshaft.Client.MAX_GET_SIZE = 2033;
...
-  // TODO: We will need to instantiate a new cdb.windshaft.PublicMap() here
-  var windshaftMap = new cdb.windshaft.PublicMap();
+  var dashboardInstance = new cdb.windshaft.DashboardInstance();
 
   var options = {
     success: function(data) {
       if (data.errors) {
-        callback(null, data);
+        // TODO: Error handling
+        throw data.errors;
```
Comment (alonsogarciapablo) [inline src/windshaft/client.js]: I need to improve this. @xavijam any ideas on how to handle errors here?

File: src/vis/vis.js [hunk 3]
```diff
@@ -471,57 +461,42 @@ var Vis = cdb.core.View.extend({
...
-        var v = self.addWidget(
-          d.type,
-          _.extend(
-            opts,
-            {
-              datasource: self.datasource
-            }
-          )
-        );
+          var widgetModel = new cdb.geo.ui.Widget.ListModel(widgetData);
```
Comment (xavijam) [inline src/vis/vis.js]: Could we change this in order to take the proper widget model and widget view?
@javier is adding histogram widget right now :D.

Comment (alonsogarciapablo) [inline src/vis/vis.js]: Done [here](https://github.com/CartoDB/cartodb.js/commit/ecde1c4a3b498ca7a5b87605539c56c2c0f9539f). We can move that logic somewhere else.

File: src/windshaft/client.js (hunk 2 above)
Comment (xavijam) [inline src/windshaft/client.js]: don't you like callbacks system?

**Comment by alonsogarciapablo** [comment]
I can use callbacks. But the thing is.. what's going to happen to the UI when this request fails? For example if a users tries to set some invalid SQL for a layer and Windshaft can't create a new instance of the dashboard, should we just throw an error in the console and leave the map/widgets as they were?

**Comment by xavijam** [comment]
Uhm... yeah, I think we should send the 'error' event and, of course, we should display that error in the widget... right?

**Comment by xavijam** [comment]
Go man go!

**Pull request merged**